.github/workflows/publish-5x-image.yml

```diff
@@ -1,17 +1,17 @@
-name: "Publish images to 4.x and 4.x-rootless tags"
+name: "Publish images to 5.x and 5.x-rootless tags"
```

Comment:
This job is used by automation for the ongoing releases, I don't think we should change it. In addition, I think I'm going to cut a new branch for this PR to land on, so it won't interfere with the versions of this image that we're actually shipping.

I'm also somewhat inclined to use a tag like `pdk3`, so we don't send the wrong message that this image can't be used with CD4PE 4.x.

Reply:
Actually, I may just leave this unmerged for now (I was thinking of calling the new branch `pdk3` anyway 😉). We can build images directly from this branch, using the job updates you've made. We just don't want them on `main`.

Reply:
OK, I didn't realise the importance of the name in that file. 
Is the 4.x related to the version of cd4pe rather than the version of the docker container? That would clear up something that's been confusing me for a while 😆. The pdk3 tag makes sense or maybe pdk3.0.1 as it might be useful at some point to have dependabot build a new version when the pdk team make a new release.

Reply:
I think the 4.x tag was originally designed to refer to the CD4PE version, yeah. That being said, we're still using that tag in 5.x too. 

When we eventually move to using a version of this image with PDK 3 by default in the product, we'll want a tag that doesn't change, since the job templates that get stored in a user's DB contain the hard-coded image tag. So if we published a new one (say, `pdk3.1.0` or whatever), any existing job templates would still be referring to `pdk3.0.1` and wouldn't pick up the new one. That's part of what's making it hard to update the default image right now to something other than `puppet/puppet-dev-tools:4.x`. You can see the default in the job template creation page: 
![image](https://github.com/puppetlabs/puppet-dev-tools/assets/839145/9b1e44c3-5606-4461-bff7-a5b2fb050f19)